test/functional/sqlsrv/sqlsrv_connect_encrypted_ksp_errors.phpt

```diff
@@ -0,0 +1,150 @@
+--TEST--
+Connect using a custom keystore provider with some required inputs missing
+--SKIPIF--
+<?php require('skipif_server_old.inc'); ?>
+--FILE--
+<?php
+
+    function connect( $server, $connectionInfo )
+    {
+        $conn = sqlsrv_connect( $server, $connectionInfo );
+        if( $conn === false )
+        {
+            echo "Failed to connect.\n";
+            $errors = sqlsrv_errors();
+            print_r( $errors[0] );
```

Comment:
this prints an array with numeric and associative indexes, which makes the expected output unnecessarily long

Reply:
how else would you suggest I check the error messages?

Reply:
print_r( $errors[0]['SQLSTATE'] );
print_r( $errors[0]['code'] );
print_r( $errors[0]['message'] );
or you can make a fancy foreach loop that asks if the key is a string then print the value.